versioned_docs/version-8.7/self-managed/identity/deployment/configuration-variables.md

```diff
@@ -132,17 +132,92 @@ steps:
+| `KEYCLOAK_CLIENTS_0_NAME`                             | A human-readable name for the client.                                 | zeebe                    |
+| `KEYCLOAK_CLIENTS_0_ID`                               | The client ID.                                                        | `${ZEEBE_CLIENT_ID}`     |
+| `KEYCLOAK_CLIENTS_0_SECRET`                           | The client secret.                                                    | `${ZEEBE_CLIENT_SECRET}` |
+| `KEYCLOAK_CLIENTS_0_TYPE`                             | The type of client, one of `CONFIDENTIAL`, `PUBLIC`, or `M2M`.        | `M2M`                    |
+| `KEYCLOAK_CLIENTS_0_PERMISSIONS_0_RESOURCE_SERVER_ID` | The ID of the first resource that defined permissions will apply to.  | `zeebe-api`              |
+| `KEYCLOAK_CLIENTS_0_PERMISSIONS_0_DEFINITION`         | The permissions to apply to the first resource.                       | `write:*`                |
+| `KEYCLOAK_CLIENTS_0_PERMISSIONS_1_RESOURCE_SERVER_ID` | The ID of the second resource that defined permissions will apply to. | `operate-api`            |
+| `KEYCLOAK_CLIENTS_0_PERMISSIONS_1_DEFINITION`         | The permissions to apply to the second resource.                      | `write:*`                |
+| `KEYCLOAK_CLIENTS_0_PERMISSIONS_2_RESOURCE_SERVER_ID` | The ID of the third resource that defined permissions will apply to.  | `tasklist-api`           |
+| `KEYCLOAK_CLIENTS_0_PERMISSIONS_2_DEFINITION`         | The permissions to apply to the third resource.                       | `write:*`                |
+| `KEYCLOAK_CLIENTS_0_PERMISSIONS_3_RESOURCE_SERVER_ID` | The ID of the fourth resource that defined permissions will apply to. | `optimize-api`           |
+| `KEYCLOAK_CLIENTS_0_PERMISSIONS_3_DEFINITION`         | The permissions to apply to the fourth resource.                      | `write:*`                |
+| `KEYCLOAK_CLIENTS_0_PERMISSIONS_4_RESOURCE_SERVER_ID` | The ID of the fifth resource that defined permissions will apply to.  | `tasklist-api`           |
+| `KEYCLOAK_CLIENTS_0_PERMISSIONS_4_DEFINITION`         | The permissions to apply to the fifth resource.                       | `read:*`                 |
+| `KEYCLOAK_CLIENTS_0_PERMISSIONS_5_RESOURCE_SERVER_ID` | The ID of the sixth resource that defined permissions will apply to.  | `operate-api`            |
+| `KEYCLOAK_CLIENTS_0_PERMISSIONS_5_DEFINITION`         | The permissions to apply to the sixth resource.                       | `read:*`                 |
+
+When converting the above variables for use in the Helm chart, ensure multiple clients or permissions are added as a list. For example, a valid `application.yaml` configuration might look like the following:
+
+```
+keycloak:
+  clients:
+    - name: "Test Application"
+      id: test-application
+      secret: your-secret
+      type: M2M
+      permissions:
+        - resource-server-id: : zeebe-api
+          definition: write:*
+```
```

Comment:
I had to backward engineer this, and it doesn't seem like Keycloak conforms to Spring conventions. Can someone verify this is valid?

Reply:
Looks correct except for the duplicated colon in line 217.